File: javascript/leetcode/125. Valid Palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
  if (s.length === 0) return true;
  let start = 0;
  let end = s.length - 1;

  const re = /^[a-z0-9]+$/i;

  while (start < end) {
    if (re.test(s[start]) && re.test(s[end])) {
      if (s[start].toLowerCase() == s[end].toLowerCase()) {
        start++;
        end--;
      } else return false;
    }

    if (!re.test(s[start])) start++;
    if (!re.test(s[end])) end--;
  }

  return true;
};

File: go/leetcode/125. Valid Palindrome.go
import "regexp"
import "strings"

func isPalindrome(s string) bool {
    if len(s) == 0 {
        return true
    }    
    
    s = strings.ToLower(s)
    
    start := 0
    end := len(s) - 1
    var isLetter = regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString

    
    for start < end {
        if isLetter(string(s[start])) && isLetter(string(s[end]))  {
            if s[start] == s[end] {
                start++; end--;
            } else {return false}
        }
        
        if !isLetter(string(s[start])) {start++;}
        if !isLetter(string(s[end])) {end--;}
    }
    
    return true
}